modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/ContextContainerRequestFilter.java

```diff
@@ -158,20 +158,11 @@ private void _handleMessage(
 		_filterExcludedOperationIds(
 			containerRequestContext, httpServletRequest, message);
 
-		_setInstanceFields(
-			instance.getClass(), httpServletRequest, message, instance);
-	}
-
-	private void _setInstanceFields(
-			Class<?> clazz, HttpServletRequest httpServletRequest,
-			Message message, Object instance)
-		throws Exception {
+		Class<?> clazz = instance.getClass();
 
-		if (clazz == Object.class) {
-			return;
-		}
+		Class<?> superClass = clazz.getSuperclass();
```

Comment:
To be honest, I don't know if I would delete the Matija's code as I guess it is a better approach if we just want to set the context for all the cases, not only the one that has a superclass :thinking:

Reply:
I think we added some useless base clases in some cases because of this limitation, but maybe we could consider it as a good practice, just to include in the base class the JAX-RS annotations and then, deleting this new code

Reply:
The old implementation would work only if those fields were present in the "Base" class.
For example [this class](https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/resource/v1_0/HeadlessDiscoveryOpenAPIResourceImpl.java#L502) wouldn't have the correct UriInfo set.